### ISSUE TITLE
FuzzyUTest fails to build looking for AtomTable

### ISSUE THREAD

**Issue by mjsduncan** [opencog/opencog#3678]
the AtomTable has been removed, right?
```
Linking CXX executable FuzzyUTest
/usr/bin/ld: /usr/local/lib/opencog/libnlpfz.so: undefined reference to `opencog::Atom::getIncomingSet(opencog::AtomSpace*) const'
/usr/bin/ld: /usr/local/lib/opencog/libnlpfz.so: undefined reference to `opencog::AtomTable::getNumAtomsOfType(unsigned short, bool) const'
collect2: error: ld returned 1 exit status
```

**Comment by linas** [comment]
Make sure you `make clean; make install` before running the unit tests. If you look at the date `ls -la /usr/local/lib/opencog/libnlpfz.so` you'll probably see something a few years old.

**Comment by mjsduncan** [comment]
there was an old version of that file so i deleted it and completely rebuilt opencog, and now it odesn't exist:
```
ls -la /usr/local/lib/opencog/libnlpfz.so
ls: cannot access '/usr/local/lib/opencog/libnlpfz.so': No such file or directory
```
is it from the opencog repo?

**Comment by linas** [comment]
My guess is that your `libnlpfz.so` was not re-built because you are missing one of the dependencies (which have changed). My guess is that you need this:
```
got clone https://github.com/opencog/lg-atomese
cd lg-atomese
mkdir build
cd build
cmake .. ; make -j6
sudo make install
```
then got back to the opencog repot, ***re-run cmake in opencog***, and then rebuild; that should now create the missing library.

General question, tho: why do you need opencog, anyway? The only thing left in that repo are assorted chatbots and NLP stuff; everything else has been split out. So unless you are working on chatbots or NLP, you do not need anything at all from the opencog git repo ...

In particular, none of the agi-bio code, nor any of the asmoses code, nor pln, nor rocca, non of that should require anything at all from the opencog git repo... if it does, that's probably a mistake of some kind.

**Comment by mjsduncan** [comment]
gee, why would i want to install the opencog repo to run opencog?
you should make it so that test doesn't run if lg-opencog isn't installed.

**Comment by linas** [comment]
@mjsduncan 

Let me rephrase that question: which part of the opencog repo do you need? I know you're not an NLP, chatbot guy, so I am surprised that you would need anything at all from that repo. 

Please be aware: more or less everything that is left in that repo has been deprecated; all the "good stuff" that people still need and use and want has been split out into it's own repo.  What's still left is everything unloved and unwanted, it's kind of a 2nd-hand-store of unwanted items. So my question to you is: if you love and want some part of it, what is that part?

**Comment by linas** [comment]
OK. The NLP stuff in this repo is a tangled mish-mash of unrelated stuff. Someday, if/when some of it is needed, it would need to be untangled and moved to specific git repos.

FYI, There is a fixed benchmark for agi-bio located here: https://github.com/opencog/benchmark/tree/master/query-loop  It uses a subset of one of the agi-bio datasets, and runs the queries that were the bottlenecks for Xabush, back a few years ago when you pulled me in.

I'd like to recommend that you use this, or a variant of this, for your measurements.

What I learned is that, for the current AtomSpace in it's current form, the primary bottleneck appears to be the scaling of the atomic ops in the CPU.   Newer CPU's e.g. the AMD Ryzen, seem to have more atomic ops per cache, and thus scale better. It's kind of hard to find details on this, it's not in the manufacturer CPU specs.  At any rate: be careful as you measure: there will (typically) be big differences between old and new AMD, between server and desktop chips, and between Intel and AMD.  (I'm talking about scaling, so multi-threaded performance. Of course, single-threaded performance will be fastest on the newest CPU's.)  Tell the metta guys to watch out for atomic ops, they bite.

**Comment by mjsduncan** [comment]
thanks for the link.  right now i'm not interested in performance, i just want to experiment with the graph search semantics.  i'll try out ivan's graphical interface & post my results to the google group thread
